ext/fg/js/frontend.js

```diff
@@ -88,6 +87,9 @@ class Frontend extends TextScanner {
     }
 
     onRuntimeMessage({action, params}, sender, callback) {
+        const {targetPopupId} = params || {};
+        if (targetPopupId !== 'all' && targetPopupId !== this.popup.id) { return; }
```

Comment:
May be better to avoid the `'all'` special constant and check if `typeof targetPopupId !== undefined` instead. Reason: I've experimented with popup IDs being strings before, and this would prevent using 'all' as an ID, which although unlikely, adds a special handling case.

Reply:
This can also run into the same `id` uniqueness issue that I mentioned for `_listenForDisplayPrepareCompleted`.

Reply:
I can change the meaning of a missing `targetPopupId` to the same as `'all'` currently. There aren't any cases when `'all'` cannot also be sent to other frames, but if such functionality is needed, another frame specific ID can be generated.